mutation-testing/check-packages-and-shards/action.yml

```
@@ -0,0 +1,159 @@
+description: "Checks which types of packages are contained in the diffs and how many mutants there are"
+branding:
+  icon: "check"
+  color: "gray-dark"
+
+outputs:
+  run_big_packages:
+    description: "True if there are packages on `stackslib` or `stacks-node`."
+    value: ${{ steps.check_packages_and_shards.outputs.run_big_packages }}
+  big_packages_with_shards:
+    description: "True if there are more than 16 mutants on `stackslib` and `stacks-node`."
+    value: ${{ steps.check_packages_and_shards.outputs.big_packages_with_shards }}
+  run_small_packages:
+    description: "True if there are packages on other packages than `stackslib` or `stacks-node`."
+    value: ${{ steps.check_packages_and_shards.outputs.run_small_packages }}
+  small_packages_with_shards:
+    description: "True if there are more than 79 (119 if `big_packages_with_shards` is true) mutants on small packages."
+    value: ${{ steps.check_packages_and_shards.outputs.small_packages_with_shards }}
+
+runs:
+  using: "composite"
+
+  steps:
+    - name: Checkout stacks-core repo
+      uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      with:
+        fetch-depth: 0
+
+    - uses: taiki-e/install-action@ac89944b5b150d78567ab6c02badfbe48b0b55aa # v2.20.16
+      name: Install cargo-mutants
+      with:
+        tool: cargo-mutants@23.12.2 # v23.12.2
+
+    - name: Relative diff
+      shell: bash
+      run: |
+        git diff origin/${{ github.base_ref }}.. > git.diff
+
+    - name: Update git diff
+      shell: bash
+      run: |
+        input_file="git.diff"
+        temp_file="temp_diff_file.diff"
+
+        # Check if the commands exist on the host
+        for cmd in tac awk sed; do
+          command -v "${cmd}" > /dev/null 2>&1 || { echo "Missing command: ${cmd}" && exit 1; }
+        done
+
+        # Check that the file exists before performing actions on it
+        if [ ! -s "$input_file" ]; then
+          echo "Diff file missing!"
+          exit 1
+        fi
+
+        # Reverse the file, remove 4 lines after '+++ /dev/null', then reverse it back (editors can't go backwards - to remove lines above)
+        tac "$input_file" > "$temp_file" && mv "$temp_file" "$input_file"
+        sed '/+++ \/dev\/null/{n;N;N;N;d;}' "$input_file" > "$temp_file" && mv "$temp_file" "$input_file"
+        tac "$input_file" > "$temp_file" && mv "$temp_file" "$input_file"
+
+        # Remove the lines between '+++ /dev/null' (included) and 'diff --git a/'
+        awk '
+          BEGIN { in_block=0 }
+          /\+\+\+ \/dev\/null/ { in_block=1; next }
+          in_block && /diff --git a\// { in_block=0; print; next }
+          !in_block
+        ' "$input_file" > "$temp_file"
+
+        # Check that the file exists before performing actions on it
+        if [ -s "$temp_file" ]; then
+          mv "$temp_file" "$input_file"
+        else
+          echo "Temp diff file missing!"
+          exit 1
+        fi
+
+        exit 0
```

small nit - this `else` is not needed. if the file exists, we can exit 0 or default to exit 1
```suggestion
        # Check that the file exists before performing actions on it
        if [ -s "$temp_file" ]; then
          mv "$temp_file" "$input_file"
          exit 0
        fi
        echo "Temp diff file missing!"
        exit 1
```